sentry_sdk/scrubber.py

```diff
@@ -77,24 +77,22 @@ def scrub_list(self, lst):
             return
 
         for v in lst:
-            if isinstance(v, dict):
```

Comment:
Note that none of these `isinstance` checks are necessary, because the `scrub_dict` and `scrub_list` methods already perform these checks internally.

Reply:
I think it'd be fine to leave these tbh. Even if they're technically not required, they make the code a bit easier to read since the flow is a bit clearer without having to check out the `scrub_x` methods. But also fine if you want to remove this.

Reply:
I found the code was slightly more difficult to understand with the if statements since I had to think through the different conditions to parse the code, but I understand what you mean.

I think we can make the code understandable without introducing branching by adding comments to explain that the methods are no-ops unless the item has type dict or list, respectively